### PR TITLE
[Issue #2967] Update Jenkinsfile.

### PR DESCRIPTION
**Pull request by glogiotatidis**

### CODE REVIEW

**Comment by Kase33** [comment]
thank you! There are so many of you out there who makw it all possible!

File: Jenkinsfile [hunk 1]
```diff
@@ -1,62 +1,89 @@
...
+            dockerImagePush("${config.project.docker_name}:full-${GIT_COMMIT_SHORT}", "mozjenkins-docker-hub")
+            dockerImagePush("${config.project.docker_name}:full-no-locales-${GIT_COMMIT_SHORT}", "mozjenkins-docker-hub")
+            dockerImagePush("${config.project.docker_name}:locales-${GIT_COMMIT_SHORT}", "mozjenkins-docker-hub")
+            dockerImagePush("${config.project.docker_name}:staticfiles-${GIT_COMMIT_SHORT}", "mozjenkins-docker-hub")
+            dockerImagePush("${config.project.docker_name}:base-dev-${GIT_COMMIT_SHORT}", "mozjenkins-docker-hub")
+            dockerImagePush("${config.project.docker_name}:base-${GIT_COMMIT_SHORT}", "mozjenkins-docker-hub")
+        }
 
-  milestone()
+        stage("Run Flake8") {
```
Comment (pmac) [inline Jenkinsfile]: To me it makes more sense to run these tests before pushing the images to docker hub? Is there a reason to push images that fail tests?

Comment (glogiotatidis) [reply Jenkinsfile]: Yes and no.

Ideally I agree we shouldn't be pushing images before tests pass. And that's why we don't mark images as `latest` unless tests pass.

In CircleCI we got this annoying `mysql has gone away` error msg and I changed the code to push images before running the tests anyway, so I could get an image and test locally or push directly if it's about a change that didn't require running the tests.

The problem with Kitsune tests is that they take toooo long so we need to make steps towards improving the speed (`test --parallel` with Django 1.11 or fancier things with `py.test`) so pushing early makes things available  to start the manual work  earlier.

Comment (glogiotatidis) [reply Jenkinsfile]: On second though:

- This pipeline runs on commits pushed to `mozilla/kitsune` so why not have images pushed to docker hub independently of the status of the tests? It shouldn't hurt and it's possible helpful with debugging?

Comment (pmac) [reply Jenkinsfile]: That seems fair. Proceed!

**Pull request merged**